hnn_core/viz.py

```diff
@@ -273,6 +273,18 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
     elif average:
         dpl = dpl + [average_dipoles(dpl)]
 
+    if type(dpl) == list:
+        for single_dpl in dpl:
+            if not isinstance(single_dpl, Dipole):
+                raise ValueError('Argument 1 should be of type '
+                                 'dipole or list of dipoles, '
+                                 f'but {single_dpl} is a {type(single_dpl)}')
+    else:
+        if not isinstance(dpl, Dipole):
```

Comment:
do you need to nest here? Zen of Python:

> Flat is better than nested

Reply:
do you *ever* enter this condition? See above in L271 ... it is turned into a list

Reply:
Hey @jasmainak I made all the changes. Please check it once